src/Sarif.Viewer.VisualStudio/OpenLogFileCommands.cs

```diff
@@ -168,117 +163,80 @@ private async System.Threading.Tasks.Task MenuItemCallbackAsync(object sender, E
-                        toolFormat = ToolFormat.FxCop;
-                        title = "Open FxCop XML log file";
-                        filter = "FxCop report and project files (*.xml)|*.xml";
-                        break;
-                    }
-                    case OpenFortifyFileCommandId:
-                    {
-                        toolFormat = ToolFormat.Fortify;
-                        title = "Open Fortify XML log file";
-                        filter = "Fortify log files (*.xml)|*.xml";
-                        break;
-                    }
-                    case OpenFortifyFprFileCommandId:
-                    {
-                        toolFormat = ToolFormat.FortifyFpr;
-                        title = "Open Fortify FPR log file";
-                        filter = "Fortify FPR log files (*.fpr)|*.fpr";
-                        break;
-                    }
-                    case OpenCppCheckFileCommandId:
-                    {
-                        toolFormat = ToolFormat.CppCheck;
-                        title = "Open CppCheck XML log file";
-                        filter = "CppCheck log files (*.xml)|*.xml";
-                        break;
-                    }
-                    case OpenClangFileCommandId:
-                    {
-                        toolFormat = ToolFormat.ClangAnalyzer;
-                        title = "Open Clang XML log file";
-                        filter = "Clang log files (*.xml)|*.xml";
-                        break;
-                    }
-                    case OpenAndroidStudioFileCommandId:
-                    {
-                        toolFormat = ToolFormat.AndroidStudio;
-                        title = "Open Android Studio XML log file";
-                        filter = "Android Studio log files (*.xml)|*.xml";
-                        break;
-                    }
-                    case OpenSemmleFileCommandId:
-                    {
-                        toolFormat = ToolFormat.SemmleQL;
-                        title = "Open Semmle QL CSV log file";
-                        filter = "Semmle QL log files (*.csv)|*.csv";
-                        break;
-                    }
-                    case OpenPylintFileCommand:
-                    {
-                        toolFormat = ToolFormat.Pylint;
-                        title = "Open Pylint JSON log file";
-                        filter = "Pylint log files (*.json)|*.json";
-                        break;
-                    }
-                    case OpenTSLintFileCommand:
-                    {
-                        toolFormat = ToolFormat.TSLint;
-                        title = "Open TSLint JSON log file";
-                        filter = "TSLint log files (*.json)|*.json";
-                        break;
-                    }
-                }
+                // Need two additional entries for "SARIF" which is at the beginning of the list.
```

Comment:
>two additional entries [](start = 24, length = 22)

"one additional entry"?

Reply:
And if you omit "None", the number of filters (when you include "SARIF") will actually be exactly equal to the number of fields -- and you'll want a comment to explain that.

---
In reply to: [505717540](https://github.com/microsoft/sarif-visualstudio-extension/pull/224#discussion_r505717540) [](ancestors = 505717540)